internal/martian/handler.go

```diff
@@ -89,7 +89,11 @@ func (p proxyHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	}
 	ctx := withSession(session)
 
-	outreq := req.Clone(ctx.addToContext(req.Context()))
+	cctx := ctx.addToContext(req.Context())
```

Comment:
How about `outreq` modification if needed?

Reply:
I did not want to make another copy of a request, but I can change it.

Reply:
Maybe refactor context creation to a function?